sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/UnsafeRowWriter.java

```diff
@@ -58,6 +58,10 @@ private void zeroOutPaddingBytes(int numBytes) {
     }
   }
 
+  public boolean isNullAt(int ordinal) {
+    return BitSetMethods.isSet(holder.buffer, startingOffset, ordinal);
```

Comment:
Do we ever start from the middle of an existing buffer? If not, can we move startingOffset into holder class?